js/src/offcanvas.js

```diff
@@ -0,0 +1,208 @@
+      return
+    }
+
+    const showEvent = EventHandler.trigger(this._element, Event.SHOW, { relatedTarget })
+
+    if (showEvent.defaultPrevented) {
+      return
+    }
+
+    this._isShown = true
+    this._element.classList.add(ClassName.SHOW)
+    this._element.removeAttribute('aria-hidden')
+    this._element.setAttribute('aria-expanded', true)
+    this._element.setAttribute('aria-offcanvas', true)
+
+    setTimeout(() => {
+      this._enforceFocus()
+
+      EventHandler.trigger(this._element, Event.SHOWN, { relatedTarget })
+      document.body.classList.add(ClassName.OPEN)
+    }, getTransitionDurationFromElement(this._element) + 9)
+  }
+
+  hide(relatedTarget) {
+    if (!this._isShown) {
+      return
+    }
+
+    const hideEvent = EventHandler.trigger(this._element, Event.HIDE, { relatedTarget })
+
+    if (hideEvent.defaultPrevented) {
+      return
+    }
+
+    this._isShown = false
+    this._element.classList.remove(ClassName.SHOW)
+    this._element.setAttribute('aria-hidden', true)
+    this._element.setAttribute('aria-expanded', false)
+    this._element.removeAttribute('aria-offcanvas')
+    this._element.blur()
+
+    setTimeout(() => {
+      document.body.classList.remove(ClassName.OPEN)
+      EventHandler.trigger(this._element, Event.HIDDEN, { relatedTarget })
+    }, getTransitionDurationFromElement(this._element))
+  }
+
+  // Private
+  _enforceFocus() {
+    const savedTabIndex = this._element.getAttribute('tabindex')
+    this._element.setAttribute('tabindex', '0')
+    this._element.focus()
+    this._element.setAttribute('tabindex', savedTabIndex || 1)
+  }
+
+  _handleClosing() {
+    EventHandler.on(this._element, Event.CLICK_DISMISS, Selector.DATA_DISMISS, event => this.hide(event))
+
+    EventHandler.on(document, Event.KEYDOWN_DATA_API, event => {
+      if (event.keyCode === ESCAPE_KEYCODE) {
+        this.hide(event)
+      }
+    })
+
+    EventHandler.on(document, Event.CLICK_DATA_API, event => {
+      const target = SelectorEngine.findOne(getSelectorFromElement(event.target))
+      if (!this._element.contains(event.target) && target !== this._element) {
+        this.hide(event)
+      }
+    })
+  }
+
+  // Static
+
+  static _jQueryInterface(config) {
+    return this.each(function () {
+      const data = Data.getData(this, DATA_KEY) || new OffCanvas(this)
+
+      if (typeof config === 'string') {
+        if (typeof data[config] === 'undefined') {
+          throw new TypeError(`No method named "${config}"`)
+        }
+
+        data[config](this)
+      }
+    })
+  }
+
+  static _getInstance(element) {
+    return Data.getData(element, DATA_KEY)
+  }
+}
+
+/**
+ * ------------------------------------------------------------------------
+ * Data Api implementation
+ * ------------------------------------------------------------------------
+ */
+
+EventHandler.on(document, Event.CLICK_DATA_API, Selector.DATA_TOGGLE, function (event) {
+  if (['A', 'AREA'].includes(this.tagName)) {
```

Comment:
`includes` isn't supported by the browsers we target 😟 you can use `.indexOf` instead